config/routes.rb

```diff
@@ -1,9 +1,15 @@
 Rails.application.routes.draw do
-  get 'users/new'
-  get 'carts/show'
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
   root 'products#index'
   get '/dashboard', to: 'dashboard#index'
+
+  get '/signup', to: 'users#new'
+  get '/login',   to: 'sessions#new'
+  post   '/login',   to: 'sessions#create'
+  delete '/logout',  to: 'sessions#destroy'
+
+  post 'products/add_to_cart/:id', to: 'products#add_to_cart', as: 'add_to_cart'
```

Comment:
I think you could move these 2 into the products resources:
```ruby
resources :products do
  post :add_to_card, on: :member
end
```

And if you also need the `new, create, index, etc.` methods you can do:
```ruby
resources :products, only: [:new, :index] do
  post :add_to_card, on: :member
end
```